src/parser/csql_grammar.y

```diff
@@ -2920,6 +2924,86 @@ create_stmt
 			$$ = qc;
 			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
 
+		DBG_PRINT}}
+	| CREATE		/* 1 */
+	  opt_or_replace	/* 2 */
+	  	{ push_msg (MSGCAT_SYNTAX_INVALID_CREATE_SYNONYM); }	/* 3 */
+	  opt_access_modifier	/* 4 */
+	  SYNONYM		/* 5 */
+	  simple_path_id	/* 6 */
+	  For			/* 7 */
+	  simple_path_id	/* 8 */
+	  opt_comment_spec	/* 9 */
+	  	{ pop_msg(); }
+		{{
+
+			PT_NODE *p = parser_new_node(this_parser, PT_CREATE_SYNONYM);
+
+			if (p)
+			  {
+			    p->info.create_synonym.or_replace = $2;
+			    p->info.create_synonym.access_modifier = $4;
+
+			    /* synonym_name, synonym_owner_name */
+			    PT_NODE *synonym = $6;
+			    if (synonym && synonym->node_type == PT_DOT_)
+			      {
+				assert (synonym->info.dot.arg1 && synonym->info.dot.arg2->node_type == PT_NAME);
+				assert (synonym->info.dot.arg2 && synonym->info.dot.arg2->node_type == PT_NAME);
+
+				if (synonym->info.dot.arg1 && synonym->info.dot.arg1->node_type == PT_NAME)
+				  {
+				    p->info.create_synonym.synonym_owner_name = synonym->info.dot.arg1;
+				  }
+
+				if (synonym->info.dot.arg2 && synonym->info.dot.arg2->node_type == PT_NAME)
+				  {
+				    p->info.create_synonym.synonym_name = synonym->info.dot.arg2;
+				  }
```

Comment:
I checked with assert(), do I need to check again?
It seems that you can choose only one of assert() and if statements.

Reply:
assert() can only be checked in debug mode,
assert_release() generates core during service.

There were no problems in the debug mode test, but problems may occur during service.

I think that null or type check is necessary when accessing a pointer variable to prevent problems that may occur during service.